ImageFunctions/Thumbnail.cs

```diff
@@ -0,0 +1,117 @@
+// Default URL for triggering event grid function in the local environment.
+// http://localhost:7071/runtime/webhooks/EventGrid?functionName={functionname}
+
+// For local testing:
+// https://{ID}.ngrok.io/runtime/webhooks/EventGrid?functionName=Thumbnail
+
+using Microsoft.Azure.EventGrid.Models;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.EventGrid;
+using Microsoft.Extensions.Logging;
+using Microsoft.WindowsAzure.Storage;
+using Microsoft.WindowsAzure.Storage.Blob;
+using Newtonsoft.Json.Linq;
+using SixLabors.ImageSharp;
+using SixLabors.ImageSharp.Formats;
+using SixLabors.ImageSharp.Formats.Gif;
+using SixLabors.ImageSharp.Formats.Jpeg;
+using SixLabors.ImageSharp.Formats.Png;
+using SixLabors.ImageSharp.PixelFormats;
+using SixLabors.ImageSharp.Processing;
+using System;
+using System.IO;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+
+namespace ImageFunctions
+{
+    public static class Thunbnail
+    {
+        private static readonly string BLOB_STORAGE_CONNECTION_STRING = Environment.GetEnvironmentVariable("AzureWebJobsStorage");
+
+        private static string GetBlobNameFromUrl(string bloblUrl)
+        {
+            var uri = new Uri(bloblUrl);
+            var cloudBlob = new CloudBlob(uri);
+            return cloudBlob.Name;
+        }
+
+        private static IImageEncoder GetEncoder(string extension)
+        {
+            IImageEncoder encoder = null;
+
+            extension = extension.Replace(".", "");
+
+            var isSupported = Regex.IsMatch(extension, "gif|png|jpe?g", RegexOptions.IgnoreCase);
+
+            if (isSupported)
+            {
+                switch (extension)
+                {
+                    case "png":
+                        encoder = new PngEncoder();
+                        break;
+                    case "jpg":
+                        encoder = new JpegEncoder();
+                        break;
+                    case "jpeg":
+                        encoder = new JpegEncoder();
+                        break;
+                    case "gif":
+                        encoder = new GifEncoder();
+                        break;
+                    default:
+                        break;
+                }
+            }
+
+            return encoder;
+        }
+
+        [FunctionName("Thumbnail")]
+        public static async Task Run(
+            [EventGridTrigger]EventGridEvent eventGridEvent,
+            [Blob("{data.url}", FileAccess.Read)] Stream input,
+            ILogger log)
+        {
+            try
```

Comment:
I love that we added some error handling to this topic. I chatted with Jeff about this, and he said that:
> yep. Throwing should result in an exception that shows up in App Insights exception table as well - and mark the execution as "Failed" so you can manually log the error or just throw and we'll log it for you exception...I guess throwing will impact the status of the execution catching and logging will just log exception so catch/log may be useful if it's a "handled" exception where the run still == successful

Just as an FYI (probably we should also document this somewhere) 😁